source/draw/RasterYuv420Semiplanar.ooc

```diff
@@ -92,30 +92,36 @@ RasterYuv420Semiplanar: class extends RasterImage {
 		}
 		result
 	}
-	resizeInto: func (target: This) {
+	resizeInto: func (target: This, sourceBox := IntBox2D new()) {
+		if (sourceBox hasZeroArea)
+			sourceBox = IntBox2D new(this size)
+		version(safe)
+			raise(sourceBox intersection(IntBox2D new(this size)) != sourceBox, "invalid source box in RasterYuv420Semiplanar resizeInto !")
+		thisSizeY := sourceBox height
+		thisSizeX := sourceBox width
 		thisYBuffer := this y buffer pointer
 		targetYBuffer := target y buffer pointer
 		for (row in 0 .. target size y) {
-			srcRow := (this size y * row) / target size y
+			srcRow := (thisSizeY * row) / target size y + sourceBox top
 			thisStride := srcRow * this y stride
 			targetStride := row * target y stride
 			for (column in 0 .. target size x) {
-				srcColumn := (this size x * column) / target size x
+				srcColumn := (thisSizeX * column) / target size x + sourceBox left
 				targetYBuffer[column + targetStride] = thisYBuffer[srcColumn + thisStride]
 			}
 		}
 		targetSizeHalf := target size / 2
-		thisSizeHalf := this size / 2
+		thisSizeHalf := sourceBox size / 2
 		thisUvBuffer := this uv buffer pointer as ColorUv*
 		targetUvBuffer := target uv buffer pointer as ColorUv*
 		if (target size y isOdd)
 			targetSizeHalf = IntVector2D new(targetSizeHalf x, targetSizeHalf y + 1)
 		for (row in 0 .. targetSizeHalf y) {
-			srcRow := (thisSizeHalf y * row) / targetSizeHalf y
+			srcRow := (thisSizeHalf y * row) / targetSizeHalf y + sourceBox top / 2
 			thisStride := srcRow * this uv stride / 2
 			targetStride := row * target uv stride / 2
 			for (column in 0 .. targetSizeHalf x) {
-				srcColumn := (thisSizeHalf x * column) / targetSizeHalf x
+				srcColumn := (thisSizeHalf x * column) / targetSizeHalf x + sourceBox left / 2
```

Comment:
Oh, I should ask: You are reading the `sourceBox` properties once per iteration, do you want to speed them up as you did with `thisSize*` and `targetSize*`?